website/server/controllers/api-v3/groups.js

```diff
@@ -137,7 +138,13 @@ api.createGroup = {
       user.party._id = group._id;
     }
 
-    const results = await Promise.all([user.save(), group.save()]);
+    const session = await mongoose.startSession();
+    let results;
+    await session.withTransaction(async () => {
+      results = await Promise.all([user.save(), group.save()]);
+    });
+    session.endSession();
```

Comment:
Don't you need to explicitly commit the transaction?

```javascript
await session.commitTransaction();
```

Reply:
Not according to the documentation.  https://mongoosejs.com/docs/transactions.html

lmk if I am misunderstanding something.

`The previous examples explicitly create a transaction and commits it. In practice, you'll want to use the session.withTransaction() helper instead. The session.withTransaction() helper handles:

Creating a transaction
Committing the transaction if it succeeds
Aborting the transaction if your operation throws
Retrying in the event of a transient transaction error.`

Reply:
Ah, indeed, you're right, I had read it as an invocation of `startTransaction` instead.

It's a pity that withTransaction doesn't return the underlying value on successful commit, that could make this code a bit nicer.

Anyway, LGTM.